File: test-server/src/main/java/al/polis/appserver/dto/CourseDto.java
package al.polis.appserver.dto;

import lombok.*;

import java.util.List;

@Data
public class CourseDto {
    private Long id;
    private String code;
    private String title;
    private String description;
    private Integer year;
    private TeacherDto teacher;
    private List<StudentDto> students;
}

File: test-server/app/src/main/java/com/polis/app/model/CourseDto.kt
package com.polis.app.model

import com.google.gson.annotations.SerializedName

data class CourseDto(
    @SerializedName("id")
    val id: Long? = null,
    
    @SerializedName("code")
    val code: String,
    
    @SerializedName("title")
    val title: String,
    
    @SerializedName("description")
    val description: String,
    
    @SerializedName("year")
    val year: Int,
    
    @SerializedName("teacher")
    val teacher: TeacherDto? = null,
    
    @SerializedName("students")
    val students: List<StudentDto>? = null
) 